test/integration/swap.behavior.ts

```diff
@@ -35,35 +34,53 @@ describe("Integration: Test Synthswap.sol", function () {
             SYNTHETIX_PROXY,
             SUSD_PROXY,
             ethers.constants.AddressZero,
-            AGGREGATION_ROUTER_V3
+            AGGREGATION_ROUTER_V3,
+            WETH
         );
         await synthswap.deployed();
 
-        // Replace 0xaaa... placeholder from generated payload with deployed Synthswap address.
-        // This is because when generating the 1inch payload we need to specify a destination receiver address,
-        // which is our Synthswap contract, and this is not know ahead of time.
-        const TRANSACTION_PAYLOAD_1INCH_WITH_DEST_AS_CONTRACT =
-            TRANSACTION_PAYLOAD_1INCH.replace(
-                /aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa/g,
-                synthswap.address.substring(2) // Slice off "0x"
-            );
-
         const signer = await ethers.getSigner(TEST_ADDRESS);
+        const chain_id = "";
+        const amount = "10000000000000000";
+        const slippage = 1;
+
+        const endpoint_1inch = 
+                "https://api.1inch.exchange/v3.0/" + chain_id 
+                + "/swap?fromTokenAddress=" + WETH
+                + "&toTokenAddress=" + SUSD
+                + "&amount=" + amount
+                + "&fromAddress=" + TEST_ADDRESS
+                + "&slippage=" + slippage
+                + "&destReceiver= " + TEST_ADDRESS
+                + "disableEstimate=true'";
+
+        // Resulting tx.data from above API call
+        // 1inch said to get mock data without actually executing a swap, set "disableEstimate: true".
+        // "https://api.1inch.exchange/v3.0/1/swap?fromTokenAddress=0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2&toTokenAddress=0x57Ab1ec28D129707052df4dF418D58a2D46d5f51&amount=10000000000000000&fromAddress=0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266&slippage=3&destReceiver=0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266&disableEstimate=true"
+        // TEST_ADDRESS is set to accounts[0] on forked main network
+        const call_data = 
+            "0x2e95b6c8000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2"
+            + "000000000000000000000000000000000000000000000000002386f26fc1000000000000"
+            + "0000000000000000000000000000000000000001c3236980d23fefdb0000000000000000"
+            + "000000000000000000000000000000000000000000000080000000000000000000000000"
+            + "000000000000000000000000000000000000000180000000000000003b6d0340f1f85b2c"
+            + "54a2bd284b1cf4141d64fd171bd85539";
 
         await synthswap
-            .connect(signer) // Call swap from TEST_ADDRESS
-            .swapInto(
-                TRANSACTION_PAYLOAD_1INCH_WITH_DEST_AS_CONTRACT,
-                SETH_BYTES32,
+            .connect(signer)    // Call swap from TEST_ADDRESS
+            .swapIntoWithETH(
+                SETH_BYTES32,   // bytes32 destinationSynthCurrencyKey
+                970000,         // uint minOut (3% slippage)
+                call_data,      // bytes calldata _data
```

Comment:
Using `call_data` with WETH input will fail in prod because it will check the user balance for WETH when calling the `swap/` endpoint. If you're using `swapIntoWithETH` it's assumed you have an ETH balance not WETH.

Reply:
swapIntoWithETH() will take msg.value (ETH), wrap it, _then_ use the call data with WETH as input and sUSD as output. Is that not the correct way to do this?